app/controllers/concerns/import.rb

```diff
@@ -36,16 +36,25 @@ def import
     end
   end
   
-  def remove_codes
-    flag = offeror_codes.where(:user_id => nil)
-    if flag.count == 0
+  def remove_unclaimed_codes
+    if offeror_codes.where(user_id: nil).count.zero?
       redirect_to "/#{params[:controller]}/home",
         :flash => { :error => "There are no unclaimed codes" }
     else
-      contents = current_offeror.remove_unclaimed_codes(offeror_codes)
-      send_data contents, :filename => "unclaimed_codes.txt"
+      current_offeror.remove_unclaimed_codes(offeror_codes)
+      redirect_to "/#{params[:controller]}/home",
+        :flash => { :notice => "Codes were removed" }
     end
   end
+  
+  def download_unclaimed_codes
+    unclaimed_codes = current_offeror.download_unclaimed_codes(offeror_codes)
+    send_data(unclaimed_codes, filename: "unclaimed_codes.txt")
+  end
+  
+  def view_codes
+    @offeror_codes = offeror_codes.all
```

Comment:
Though the View Codes button (invoking this action) is not rendered when neither Redemify nor vendor codes persist in the db, the nil object bug can be reproduced by submitting the GET 'vendors/view_codes' or GET 'providers/view_codes' request via the browser address bar and the codes missing in the persistence tier.
The codes availability check and the redirect_to statement should likely be appropriate.